modules/core/src/main/java/org/apache/ignite/internal/management/persistence/PersistenceAbstractCommand.java

```diff
@@ -0,0 +1,122 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.management.persistence;
+
+import java.util.List;
+import java.util.function.Consumer;
+import org.apache.ignite.internal.management.api.ComputeCommand;
+import org.apache.ignite.internal.management.persistence.PersistenceCommand.PersistenceCleanAllTaskArg;
+import org.apache.ignite.internal.management.persistence.PersistenceCommand.PersistenceCleanCorruptedTaskArg;
+import org.apache.ignite.internal.management.persistence.PersistenceCommand.PersistenceTaskArg;
+import org.apache.ignite.internal.visor.persistence.PersistenceTask;
+import org.apache.ignite.internal.visor.persistence.PersistenceTaskResult;
+import org.apache.ignite.lang.IgniteBiTuple;
+import static org.apache.ignite.internal.management.api.CommandUtils.INDENT;
+
+/** */
+public abstract class PersistenceAbstractCommand implements ComputeCommand<PersistenceTaskArg, PersistenceTaskResult> {
+    /** {@inheritDoc} */
+    @Override public Class<PersistenceTask> taskClass() {
+        return PersistenceTask.class;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void printResult(PersistenceTaskArg arg, PersistenceTaskResult res, Consumer<String> printer) {
+        if (!res.inMaintenanceMode())
+            printer.accept("Persistence command can be sent only to node in Maintenance Mode.");
+        //info command
+        else if (res.cachesInfo() != null) {
+            printer.accept("Persistent caches found on node:");
+
+            //sort results so corrupted caches occur in the list at the top
+            res.cachesInfo().entrySet().stream().sorted((ci0, ci1) -> {
+                IgniteBiTuple<Boolean, Boolean> t0 = ci0.getValue();
+                IgniteBiTuple<Boolean, Boolean> t1 = ci1.getValue();
+
+                boolean corrupted0 = t0.get1() || t0.get2();
+                boolean corrupted1 = t1.get1() || t1.get2();
+
+                if (corrupted0 && corrupted1)
+                    return 0;
+                else if (!corrupted0 && !corrupted1)
+                    return 0;
+                else if (corrupted0 && !corrupted1)
+                    return -1;
+                else
+                    return 1;
+            }).forEach(
+                e -> {
+                    IgniteBiTuple<Boolean, Boolean> t = e.getValue();
+
+                    String status;
+
+                    if (!t.get1())
```

Comment:
It seems that there is a bug (the same condition in this `if` and the consequent `else if`

Reply:
Fixed